protocols/v2/codec-sv2/src/lib.rs

```diff
@@ -1,3 +1,60 @@
+//! # Stratum V2 Codec Library
+//!
+//! This crate provides the encoding and decoding functionality for the Stratum V2 protocol,
+//! handling secure communication between clients and servers.
+//!
+//! `codec-sv2` is essential for implementing the Sv2 protocol correctly and securely. It abstracts
+//! the complexity of message encoding, decoding, and encryption, providing a reliable and
+//! consistent foundation for mining software imperative for interoperability.
+//!
+//! ## Features
+//!
+//! * **Encoding/Decoding**: Serialize and deserialize Stratum V2 messages.
+//! * **Noise Protocol**: Support for secure communication using the Noise protocol framework.
+//! * **Error Handling**: Comprehensive error handling for various stages of communication and
+//!   encryption.
+//!
+//! ## Modules
+//!
+//! * `decoder`: Provides functionality to decode Stratum V2 messages.
+//! * `encoder`: Provides functionality to encode Stratum V2 messages.
+//! * `error`: Defines error types and result aliases used throughout the crate.
```

Comment:
a `#Modules` section is automatically generated by Rust docs, so it is redundant to add one here

laos, the `encoder` and `decoder` modules are not exposed

Reply:
assuming we will make `encoder` and `decoder` modules public, we should still remove this `## Modules` section because it's redundant